src/main/kotlin/com/openlattice/assembler/AssemblerConnectionManager.kt

```diff
@@ -75,6 +79,10 @@ class AssemblerConnectionManager(
         metricRegistry: MetricRegistry
 ) {
 
+    private val perDbCache: LoadingCache<String, HikariDataSource> = CacheBuilder
+            .newBuilder()
+            .expireAfterAccess(1, TimeUnit.HOURS)
```

Comment:
I'm not sure, we should let it expire, since we use 1 connection per organization and as far as I know, organizations don't change a lot. As an alternative, we could remove the expiration and remove it manually, when an organization is deleted. 
In that case, we should look out for the order of organization deletion handling (`OrganizationsController.destroyOrganization`) in `Assembler` and `ExternalDatabaseManagementService`

Reply:
I'm open to this approach.

For services like this I prefer to avoid tight coupling and would rather free unused resources optimistically. We've already survived allocating new connection pools every few seconds, so once an hour should be ok.

Reply:
That's true, if you think this approach is better, I'm ok with that too.